gse-util/src/main/java/com/powsybl/gse/util/NodeChooser.java

```diff
@@ -807,32 +807,23 @@ private void savePreferences() {
     public static <N, F extends N, D extends N, T extends N> Optional<T> showAndWaitDialog(
             TreeModel<N, F, D> treeModel, Window window, AppData appData, GseContext context,
             BiPredicate<N, TreeModel<N, F, D>> filter, Set<String>... openedProjectsList) {
-        Dialog<T> dialog = new Dialog<>();
+        Dialog<T> dialog = null;
         try {
-            dialog.setTitle(RESOURCE_BUNDLE.getString("OpenFile"));
-            dialog.getDialogPane().getButtonTypes().addAll(ButtonType.OK, ButtonType.CANCEL);
             NodeChooser<N, F, D, T> nodeChooser = new NodeChooser<>(window, treeModel, appData, context, filter, openedProjectsList);
+            dialog = new GseDialog<>(RESOURCE_BUNDLE.getString("OpenFile"), nodeChooser, window, nodeChooser.selectedNodeProperty().isNull(), buttonType -> buttonType == ButtonType.OK ? nodeChooser.selectedNodeProperty().get() : null);
```

Comment:
Extract the lambda used as a Callback to a variable to improve readability.

Reply:
updated